apps/avatax/src/pages/api/webhooks/checkout-calculate-taxes.ts

```diff
@@ -50,22 +104,54 @@ export default withOtel(
     logger.info("Handler for CHECKOUT_CALCULATE_TAXES webhook called");
 
     try {
-      verifyCalculateTaxesPayload(payload);
-      logger.debug("Payload validated Successfully");
+      const payloadVerificationResult = verifyCalculateTaxesPayload(payload);
+
+      payloadVerificationResult
+        .mapErr((error) => {
+          logger.error(`[${error.name}] ${error.message}`);
+          Sentry.captureException("Invalid payload from Saleor was sent to Avatax app");
+
+          return webhookResponse.error(error);
+        })
+        .map(() => {
+          logger.debug("Payload validated Successfully");
+        });
 
       const appMetadata = payload.recipient?.privateMetadata ?? [];
       const channelSlug = payload.taxBase.channel.slug;
-      const activeConnectionService = getActiveConnectionService(
+      const activeConnectionServiceResult = getActiveConnectionService(
         channelSlug,
         appMetadata,
         ctx.authData,
       );
 
-      logger.info("Found active connection service. Calculating taxes...");
-      const calculatedTaxes = await activeConnectionService.calculateTaxes(payload);
+      if (activeConnectionServiceResult.isErr()) {
+        const err = activeConnectionServiceResult.error;
+
+        logger.debug(`Error in taxes calculation occured: ${err.name} ${err.message}`, {
+          error: err,
+        });
+
+        const errorStrategy = webhookErrorToResponseMapper(req, res).get(err.name);
+
+        if (errorStrategy) {
+          return errorStrategy();
+        } else {
+          Sentry.captureException(err);
+          logger.fatal(`UNHANDLED: ${error.name}`, {
+            error: error,
+          });
+
+          return res.status(500).send("Error calculating taxes");
+        }
```

Comment:
suggestion: I think this could be extracted, because we will have the same logic for every webhook:

1. use error to response mapper
2. return a fallback

Reply:
Yes, but I will do this later, to avoid large code changes.